apis/v1alpha2/gateway_types.go

```diff
@@ -663,19 +651,24 @@ const (
 
 // ListenerStatus is the status associated with a Listener.
 type ListenerStatus struct {
-	// Port is the unique Listener port value for which this message is
-	// reporting the status.
-	Port PortNumber `json:"port"`
-
-	// Protocol is the Listener protocol value for which this message is
-	// reporting the status.
-	Protocol ProtocolType `json:"protocol"`
+	// Name is the name of the Listener.
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
 
-	// Hostname is the Listener hostname value for which this message is
-	// reporting the status.
+	// SupportedKinds is the list indicating the Kinds supported by this
+	// listener. When this is not specified on the Listener, this MUST represent
+	// the kinds an implementation supports for the specified protocol. When
+	// there are kinds specified on the Listener, this MUST represent the
+	// intersection of those kinds and the kinds supported by the implementation
+	// for the specified protocol.
 	//
-	// +optional
-	Hostname *Hostname `json:"hostname,omitempty"`
+	// +kubebuilder:validation:MaxItems=8
+	SupportedKinds []RouteGroupKind `json:"supportedKinds,omitempty"`
+
+	// AttachedRoutes represents the total number of Routes that have been
+	// successfully attached to this Listener.
```

Comment:
```suggestion
	// successfully attached to this Listener.
	//
	// +kubebuilder:validation:Minimum=1 
```

Reply:
Isn't it valid for this to be 0? No guarantee that a Route will be attached to a Listener.